src/components/Checkbox/index.tsx

```diff
@@ -247,20 +239,19 @@ const Checkbox = forwardRef(
             <Loader active size={size} />
           </StyledActivityContainer>
         ) : null}
-        <StyledCheckBoxContainer
+        <CheckboxContainer
           className={className}
           aria-disabled={disabled}
           data-visibility={dataVisibility}
           data-checked={checked}
           data-error={!!error}
         >
-          <StyledReakitCheckbox
+          <CheckboxInput
+            type="checkbox"
             aria-invalid={!!error}
             aria-describedby={error ? `${computedName}-hint` : undefined}
-            aria-checked={
-              state === 'indeterminate' ? 'mixed' : (state as boolean)
-            }
-            checked={state === 'indeterminate' ? false : (state as boolean)}
+            aria-checked={state === 'indeterminate' ? 'mixed' : undefined}
```

Comment:
I'm not sure about this, but I think it should behave the same as `checked`?
```suggestion
            aria-checked={state === 'indeterminate' ? 'mixed' : state}
```

Reply:
@QuiiBz the best is not to use `aria-checked` as it is for a11y. In that case we need it only for `mixed` state as it is the only way to differentiate for true/false state :)